src/services/auth.service.ts

```diff
@@ -79,5 +80,24 @@ export const authService = {
 
     isMobile() {
         return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
+    },
+
+    getAuthTokenExpiration(): ITokenExpiration {
+        const jwtDecode = require('jwt-decode');
+        const token = getAuthToken();
+        const decoded = jwtDecode(token);
+
+        // no higher than refreshInterval in seconds
+        const refreshOffset = 15;
```

Comment:
I would actually set up that the offset as an ENV variable so we can actually control that without having to hardcode it in the source. What do you think?

Reply:
I agree, it's a better approach. Fixed.